core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -862,6 +962,25 @@ private class ShuffleFetchCompletionListener(var data: ShuffleBlockFetcherIterat
 private[storage]
 object ShuffleBlockFetcherIterator {
 
+  /**
+   * A flag which indicates whether the Netty OOM error has raised during shuffle.
+   * If true, unless there's no in-flight fetch requests, all the pending shuffle
+   * fetch requests will be deferred until the flag is unset (whenever there's a
+   * complete fetch request).
+   */
+  val isNettyOOMOnShuffle = new AtomicBoolean(false)
+
+  /**
+   * The max number of a block could retry due to Netty OOM before throwing the fetch failure.
+   */
+  val maxAttemptsOnNettyOOM: Int = SparkEnv.get.conf.get(SHUFFLE_MAX_ATTEMPTS_ON_NETTY_OOM)
```

Comment:
It's a bit fragile to read a spark conf in `object`. Shall we read it when we instantiate `ShuffleBlockFetcherIterator`?